DMCompiler/DM/Visitors/DMObjectBuilder.cs

```diff
@@ -72,11 +72,38 @@ class DMObjectBuilder {
 
             _currentObject = DMObjectTree.GetDMObject(varDefinition.ObjectPath);
 
-            if (varDefinition.IsGlobal) {
-                variable = _currentObject.CreateGlobalVariable(varDefinition.Type, varDefinition.Name, varDefinition.IsConst);
-            } else {
-                variable = new DMVariable(varDefinition.Type, varDefinition.Name, false, varDefinition.IsConst);
-                _currentObject.Variables[variable.Name] = variable;
+            //DMObjects store two bundles of variables; the statics in GlobalVariables and the non-statics in Variables.
+            if (varDefinition.IsStatic) { // static
+
+                //make sure this static doesn't already exist first
+                if(DoesOverrideGlobalVars(varDefinition)) // Some snowflake behaviour for global.vars
+                {
+                    DMCompiler.Error(new CompilerError(varDefinition.Location, "Duplicate definition of global.vars"));
+                    //We can't salvage any part of this definiton, since global.vars doesn't technically even exist, so lets just return
+                    _currentObject = oldObject; //TODO: Does C# have deferred things, like with destructor fanciness or something?
+                    return;
+                }
+                if (_currentObject.GlobalVariables.ContainsKey(varDefinition.Name))
+                {
+                    DMCompiler.Error(new CompilerError(varDefinition.Location, $"Duplicate definition of static var {varDefinition.Name}"));
```

Comment:
Wrapping the name of the var in quotes would make the message a little clearer
```suggestion
                    DMCompiler.Error(new CompilerError(varDefinition.Location, $"Duplicate definition of static var \"{varDefinition.Name}\""));
```